src/app/core/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,60 @@
+import { forwardRef, Ref, type ReactNode } from "react";
+import clsx from "clsx";
+import { LinkIcon } from "@/app/core/components/Icons/LinkIcon";
+
+export type TBadgeSize = "regular";
+
+interface IBadgeProps {
+  children: ReactNode;
+  icon: ReactNode;
+  outLink?: string;
+  size?: TBadgeSize;
+}
+
+export const BadgeSizes = {
+  regular: "px-1 pe-2 py-1 text-sm",
+};
+
+export const BadgeStyles =
+  "flex rounded-full items-center bg-transparent border border-gray-400 text-gray-700 dark:text-gray-300";
+
+const Badge = forwardRef(
+  (
+    { children, size = "regular", icon, outLink = "" }: IBadgeProps,
+    ref: Ref<HTMLDivElement>
+  ) => {
+    const Wrapper = outLink ? "a" : "div";
+
+    return (
+      <div ref={ref}>
+        <Wrapper
+          href={outLink || undefined}
+          target={outLink ? "_blank" : undefined}
+          rel={outLink ? "noopener noreferrer" : undefined}
+          className={clsx(
+            "font-bold transition-all",
+            outLink
+              ? "hover:text-breadpink-500 hover:border-breadpink-500 dark:hover:text-breadpink-shaded dark:hover:border-breadpink-shaded cursor-pointer"
+              : "",
+            BadgeSizes[size],
+            BadgeStyles
+          )}
+        >
+          {icon && <span className="flex items-center me-2">{icon}</span>}
+          <span className="dark:text-white text-gray-700">{children}</span>
+          {outLink && (
+            <span id="linkIcon" className="flex ms-auto md:ms-2">
+              <a href={outLink} className="">
+                <LinkIcon />
+              </a>
```

Comment:
You shouldn't nest an anchor element inside another anchor element as it's not valid HTML:

https://stackoverflow.com/questions/13052598/creating-anchor-tag-inside-anchor-tag